### PR TITLE
De 910 contacts

### PR DESCRIPTION
**Pull request by hanssonandreas**
# Description of change
(write a short description or paste a link to JIRA)

# Manual QA steps
 - 
 
# Risks
 - 
 
# Rollback steps
 - revert this branch

### CODE REVIEW

**Review by paddie — CHANGES_REQUESTED** [review]

File: tap_hubspot/hubspot.py [hunk 1]
```diff
@@ -182,6 +182,44 @@ def streams(self, start_date: datetime, end_date: datetime):
         else:
             raise NotImplementedError(f"unknown stream_id: {self.tap_stream_id}")
 
+    def get_contacts(
+        self, start_date: datetime, end_date: datetime
+    ) -> Iterable[Tuple[Dict, datetime]]:
+        filter_key = "lastmodifieddate"
+        obj_type = "contacts"
+
+        properties = self.get_object_properties(obj_type)
```
Comment (paddie) [inline tap_hubspot/hubspot.py:190]: remember to delete the top-level `contacts` entry in the `MANDATORY_PROPERTIES`

File: tap_hubspot/hubspot.py [hunk 2]
```diff
@@ -182,6 +182,44 @@ def streams(self, start_date: datetime, end_date: datetime):
...
+
+        gen = self.search(
+            obj_type,
+            filter_key,
+            start_date,
+            end_date,
+            properties,
+        )
+
+        for chunk in chunker(gen, 10):
```
[the remark this answers is not in the document]
Comment (paddie) [reply tap_hubspot/hubspot.py:200]: would use 100

File: tap_hubspot/hubspot.py [hunk 3]
```diff
@@ -182,6 +182,44 @@ def streams(self, start_date: datetime, end_date: datetime):
...
+            ids: List[str] = [contact["id"] for contact in chunk]
+
+            deals_associations = self.get_associations(obj_type, "deals", ids)
```
[the remark this answers is not in the document]
Comment (hanssonandreas) [reply tap_hubspot/hubspot.py:203]: So we shall not get the deals associations for contact endpoint?

File: tap_hubspot/hubspot.py [hunk 4]
```diff
@@ -182,6 +182,44 @@ def streams(self, start_date: datetime, end_date: datetime):
...
+            companies_associations = self.get_associations(obj_type, "companies", ids)
+
+            for i, contact_id in enumerate(ids):
+                contact = chunk[i]
+
+                deals = deals_associations.get(contact_id, [])
+                companies = companies_associations.get(contact_id, [])
+
+                contact["associations"] = {
+                    "contacts": {"results": deals},
```
[the remark this answers is not in the document]
Comment (hanssonandreas) [reply tap_hubspot/hubspot.py]: Same question here. Should we only get companies associations for contacts?

File: tap_hubspot/hubspot.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (hanssonandreas) [reply tap_hubspot/hubspot.py:200]: yes

File: tap_hubspot/hubspot.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (paddie) [reply tap_hubspot/hubspot.py]: test

File: tap_hubspot/hubspot.py [hunk 5]
```diff
@@ -172,7 +172,7 @@ def get_deals_v2(
         )
 
         for chunk in chunker(gen, 100):
-            ids: List[str] = [contact["id"] for contact in chunk]
+            ids: List[str] = [deal["id"] for deal in chunk]
```
Comment (paddie) [inline tap_hubspot/hubspot.py:175]: this got introduced by this PR for some reason

File: tap_hubspot/hubspot.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (paddie) [reply tap_hubspot/hubspot.py]: for now, yes

[the remark this answers is not in the document]
Comment (hanssonandreas) [reply tap_hubspot/hubspot.py]: This has been fixed

File: tap_hubspot/hubspot.py (hunk 5 above)
Comment (hanssonandreas) [reply tap_hubspot/hubspot.py:175]: Has been changed in latest commit on this PR

File: tap_hubspot/hubspot.py [hunk 6]
```diff
@@ -389,21 +427,6 @@ def get_companies(self):
...
-        offset_key = "after"
-        replication_path = ["updatedAt"]
-        params = {"limit": 100, "properties": MANDATORY_PROPERTIES["contacts"]}
-        yield from self.get_records(
-            path,
-            replication_path,
-            params=params,
-            data_field=data_field,
-            offset_key=offset_key,
-        )
```
Comment (paddie) [inline tap_hubspot/hubspot.py:406]: I would keep this as the very last commit - just so we have a reference for what the old thing did for the rest of the commits

File: tap_hubspot/hubspot.py [hunk 7]
```diff
@@ -237,7 +236,7 @@ def get_deals_v2(
         )
 
         for chunk in chunker(gen, 10):
-            ids: List[str] = [deal["id"] for deal in chunk]
+            ids: List[str] = [contact["id"] for contact in chunk]
```
Comment (paddie) [inline tap_hubspot/hubspot.py]: could you do a fixup for this one to clean up the history?

File: tap_hubspot/hubspot.py [hunk 8]
```diff
@@ -197,20 +137,17 @@ def get_contacts(
             properties,
         )
 
-        for chunk in chunker(gen, 10):
+        for chunk in chunker(gen, 100):
```
Comment (paddie) [inline tap_hubspot/hubspot.py:139]: split this into two commits - removing the old configurations stuff is low-risk stuff, while this is important I wouldn't want this getting lost in the PRs.

File: tap_hubspot/hubspot.py [hunk 9]
```diff
@@ -197,20 +137,17 @@ def get_contacts(
...
             companies_associations = self.get_associations(obj_type, "companies", ids)
 
             for i, contact_id in enumerate(ids):
                 contact = chunk[i]
 
-                deals = deals_associations.get(contact_id, [])
                 companies = companies_associations.get(contact_id, [])
 
                 contact["associations"] = {
-                    "deals": {"results": deals},
```
Comment (paddie) [inline tap_hubspot/hubspot.py]: we should split these into separate commits as well, or fixup them into the initial one.

File: tap_hubspot/hubspot.py (hunk 5 above)
Comment (paddie) [reply tap_hubspot/hubspot.py:175]: could you rebase it to completely remove it from history?

**Review by paddie — CHANGES_REQUESTED** [review]

File: tap_hubspot/hubspot.py (hunk 7 above)
Comment (hanssonandreas) [reply tap_hubspot/hubspot.py]: This has been fixed in a later commit

**Review by paddie — APPROVED** [review]
